main.go

```diff
@@ -64,8 +174,8 @@ func (b *Backend) healthCheck() {
 	for {
 		req, err := http.NewRequest(http.MethodGet, healthCheckURL, nil)
 		if err != nil {
-			if b.logging {
-				fmt.Printf("%s %s fails\n", b.endpoint, err)
+			if globalLogFlag {
```

Comment:
This is not a flag its the value of the flag, so we should call it `globalLoggingEnabled` ?